src/App.tsx

```diff
@@ -15,35 +16,103 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabet = 'Sort alphabetically',
+  Length = 'Sort by length',
+}
+
+function goodInUse(
+  good: string[],
+  isReverse: boolean,
+  sortTypeFilter: string,
+): string[] {
+  const currentGoods: string[] = [...good];
+
+  if (sortTypeFilter) {
+    currentGoods.sort((item1, item2) => {
+      switch (sortTypeFilter) {
+        case SortType.Alphabet:
+          return item1.localeCompare(item2);
+        case SortType.Length:
+          return item1.length - item2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReverse) {
+    currentGoods.reverse();
+  }
+
+  return currentGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortTypeFilter, setSortTypeFilter] = useState('');
+  const [isReverse, setIsReverse] = useState(false);
+  const goodsList: string[] = goodInUse(
+    goodsFromServer,
+    isReverse,
+    sortTypeFilter,
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortTypeFilter !== SortType.Alphabet,
+          })}
+          onClick={() => setSortTypeFilter(SortType.Alphabet)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortTypeFilter !== SortType.Length,
+          })}
+          onClick={() => setSortTypeFilter(SortType.Length)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': !isReverse,
+          })}
+          onClick={() => setIsReverse(!isReverse)}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {sortTypeFilter === '' && isReverse === false ? (
+          ''
+        ) : (
```

Comment:
You don't need to compare with an empty string, just check if a value exists or not. And the same about comparing with boolean variables
https://developer.mozilla.org/en-US/docs/Glossary/Falsy
https://developer.mozilla.org/en-US/docs/Glossary/Truthy